### PR TITLE
Fixing ADC initialization function to work properly on Portenta H7

### PR DESCRIPTION
**Pull request by leonardocavagnis**
Main changes:
- Fixing bugs in `adc.begin()` function to allow properly working on Portenta H7
- Adding error checking in `adc.begin()` function
- Updating `README.md` according to the Arduino library guideline
- Restyling and updating `docs/readme.md`

The PR is based on the [ArduinoCore-mbed patch](https://github.com/arduino/ArduinoCore-mbed/commit/1e561292d0c8bd20d92ad78b8fa9c96d25a8501d) related to arduino/mbed-os [PR#34](https://github.com/arduino/mbed-os/pull/34).

Please,
@iabdalkader take a look to `adc.begin()` modifies
@karlsoderby check if `docs/readme.md` sounds good for you

### CODE REVIEW

File: docs/readme.md [hunk 1]
```diff
@@ -1,12 +1,168 @@
...
 
-To use this library:
+## Features
+
+- ADC/DAC parameters fine tuning: resolution, channel number, queue number and size
+- ADC acquisition with DMA in double buffering mode
+- ADC Multichannel acquisition
+- DAC Multichannel writing 
+- Storing ADC samples history in multiple queues
+
```
Comment (karlsoderby) [inline docs/readme.md:16]:
```suggestion
## Guides

To learn more about using the DAC & ADC on the GIGA R1 WiFi, check out the [GIGA Advanced DAC/ADC Guide](https://docs.arduino.cc/tutorials/giga-r1-wifi/giga-audio).

This includes examples such as audio playback from USB storage, generate sine waves and more.
```

File: docs/readme.md [hunk 2]
```diff
@@ -1,12 +1,168 @@
...
+Here is a example for the Portenta H7:
+
+```cpp
+#include <Arduino_AdvancedAnalog.h>
+
+AdvancedADC adc_c(A2, A3, A4); 
+/* Mapped to ADC3 */
+
+AdvancedADC adc_d(A5);
```
Comment (karlsoderby) [inline docs/readme.md:114]: @leonardocavagnis should `adc_c` and `adc_d` be -> `adc_a` and `adc_b`?

**Review by karlsoderby — COMMENTED** [review]
Hey @leonardocavagnis I've reviewed the documentation part and it looks great. Left some small suggestions :)

File: docs/readme.md (hunk 2 above)
Comment (leonardocavagnis) [reply docs/readme.md:114]: No, I chose them to differentiate the two examples 😉

**Comment by iabdalkader** [comment]
Hi @leonardocavagnis can you please squash all of the docs update commits into one commit ?

**Comment by iabdalkader** [comment]
@karlsoderby @leonardocavagnis Why are we removing most of the `README.md` file ?

**Comment by leonardocavagnis** [comment]
> @karlsoderby @leonardocavagnis Why are we removing most of the `README.md` file ?

I removed it according to the Arduino Library guideline. Most of the information has been moved in the` docs/README.md`

**Review by iabdalkader — CHANGES_REQUESTED** [review]

**Pull request merged**